api/backend.proto

```diff
@@ -77,15 +77,15 @@ message MatchFunctionConfig {
   }
 }
 
-message GetMatchesRequest {
+message CreateProposalRequest {
   // MatchFunction to be executed for the given list of MatchProfiles
   MatchFunctionConfig function = 1;
 
   // MatchProfiles for which this MatchFunction should be executed.
   repeated MatchProfile profile = 2;
 }
 
-message GetMatchesResponse {
+message CreateProposalResponse {
   // Result Match for the requested MatchProfile.
   Match match = 1;
```

Comment:
Should we rename Match as Proposal?
Also @Laremere had an interesting proposal to split this definition out so there's an immutable and append-only object.

It's very strange to have a create verb that returns a resource of a type that's different.